prov/efa/src/efa_mr.c

```diff
@@ -934,8 +931,14 @@ static int efa_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 	 * can be access from host. EFA provider considers all device memory
 	 * to be accessed by device only. Therefore, this function claim
 	 * support of this flag, but do not save it in efa_mr.
+	 *
+	 * FI_MR_DMABUF:
+	 * This flag indicates that the memory region to registered is
+	 * a DMA-buf backed region. When set, the region is specified through
+	 * the dmabuf field of the fi_mr_attr structure. This flag is only
+	 * usable for domains opened with FI_HMEM capability support.
 	 */
```

Comment:
Do we need to check API version?

Reply:
We need, added

Reply:
If we are not backporting, do we still need a version check?

Reply:
I think it's a different thing. The dmabuf API is only introduced in 1.20, so we don't expect user to access it when using an older Libfabric API version. They need to use FI_VERSION(1,20) in fi_getinfo to access this feature. Libfabric API version and Libfabric version are separate

Reply:
@j-xiong do you think we should be defensive for this api version check?

Reply:
Thinking about it twice, I think we need this api check, NCCL plugin will toggle dmabuf/non-dmabuf reg based on the Libfabric API version

Reply:
Yes, I think having the version check here is recommended.